x-pack/plugins/observability/public/components/custom_threshold/components/preview_chart/preview_chart.tsx

```diff
@@ -0,0 +1,322 @@
+  };
+
+  useEffect(() => {
+    if (!formulaAsync.value || !dataView || !formula) {
+      return;
+    }
+    const baseLayer = {
+      type: 'formula',
+      value: formula,
+      label: 'Custom Threshold',
+      groupBy,
+      format: {
+        id: 'number',
+        params: {
+          decimals: 0,
+        },
+      },
+      filter: {
+        language: 'kuery',
+        query: filterQuery || '',
+      },
+    };
+
+    const xYDataLayerOptions: XYLayerOptions = {
+      buckets: { type: 'date_histogram' },
+      seriesType: 'bar',
+    };
+
+    if (groupBy && groupBy?.length) {
+      xYDataLayerOptions.breakdown = {
+        type: 'top_values',
+        field: groupBy[0],
+        params: {
+          size: 3,
+          secondaryFields: groupBy as string[],
+          accuracyMode: false,
+        },
+      };
+    }
+
+    const xyDataLayer = new XYDataLayer({
+      data: [baseLayer].map((layer) => ({
+        type: layer.type,
+        value: layer.value,
+        label: layer.label,
+        format: layer.format,
+        filter: layer.filter,
+      })),
+      options: xYDataLayerOptions,
+    });
+
+    const layers: Array<XYDataLayer | XYReferenceLinesLayer> = [xyDataLayer];
+    if (thresholdReferenceLine) {
+      layers.push(...thresholdReferenceLine);
+    }
+    const attributesLens = new LensAttributesBuilder({
+      visualization: new XYChart({
+        visualOptions: {
+          valueLabels: 'hide',
+        },
+        layers,
+        formulaAPI: formulaAsync.value.formula,
+        dataView,
+      }),
+    }).build();
+    const lensBuilderAtt = { ...attributesLens, type: 'lens' };
+    setAttributes(lensBuilderAtt);
+  }, [
+    comparator,
+    dataView,
+    equation,
+    filterQuery,
+    formula,
+    formulaAsync.value,
+    groupBy,
+    metrics,
+    threshold,
+    thresholdReferenceLine,
+  ]);
+
+  if (
+    !dataView ||
+    !attributes ||
+    error?.equation ||
+    Object.keys(error?.metrics || {}).length !== 0
+  ) {
+    return (
+      <div style={{ maxHeight: 180, minHeight: 180 }}>
+        <EuiEmptyPrompt
+          iconType="visArea"
+          titleSize="xxs"
+          body={
+            <FormattedMessage
+              id="xpack.observability.customThreshold.rule..charts.noData.title"
+              defaultMessage="No chart data available, check the rule {errorSourceField}"
+              values={{
+                errorSourceField:
+                  Object.keys(error?.metrics || {}).length !== 0
+                    ? 'aggregation fields'
+                    : error?.equation
+                    ? 'equation'
+                    : 'conditions',
+              }}
+            />
+          }
+        />
+      </div>
+    );
+  }
+  return (
+    <div>
+      <lens.EmbeddableComponent
+        onLoad={setChartLoading}
+        id="customThresholdPreviewChart"
+        style={{ height: 180 }}
+        timeRange={{ from: `now-${timeSize}${timeUnit}`, to: 'now' }}
```

Comment:
@fkanout This will cover only one window in preview chart. We should add multiple windows (e.g. if rule window is `1m`, we can show data of `1m` * 5) to have a broader view of data. Even in preview chart, I think it is useful to see how data is flowing for short period.

Reply:
@benakansara, good point!

Reply:
Maybe we can use the value we had before, I think @fkanout said it was 20 * timeSize?

Reply:
Making the time range 20 times the current value made the chart look better, but still, each bar is for 30 seconds rather than 1 minute.

|Before|After|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/50685f47-a7fe-47ce-a4eb-25295a377ce1)|![image](https://github.com/elastic/kibana/assets/12370520/4bcee0fd-ce23-48fb-9ee7-7bcb50f2d086)|

Reply:
it's possible to set custom interval in lens embeddable column. that will force it to not auto interval and can match what's desired.

Reply:
@shahzad31 How can we do that?

Reply:
for example to suggest an hour interval , this is how column will look like
```
{
                                                "label": "@timestamp",
                                                "dataType": "date",
                                                "operationType": "date_histogram",
                                                "sourceField": "@timestamp",
                                                "isBucketed": true,
                                                "scale": "interval",
                                                "params": {
                                                    "interval": "h",
                                                    "includeEmptyRows": false,
                                                    "dropPartials": false
                                                }
                                            }
```

Reply:
you can play around in lens ui 
<img width="481" alt="image" src="https://github.com/elastic/kibana/assets/3505601/fb7504cf-1aac-4fa6-af1a-8d62024b1024">